bazel_tools/client_server/runner_with_port_file/Main.hs

```diff
@@ -0,0 +1,71 @@
+-- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-% LANGUAGE OverloadedStrings %-}
+
+module Main(main) where
+
+import Control.Concurrent
+import Control.Exception.Safe
+import Control.Monad (forM, forM_)
+import Control.Monad.Loops (untilJust)
+import Data.List.Split (splitOn)
+import Network.Socket
+import System.Directory (removeFile)
+import System.Environment
+import System.IO.Temp (emptySystemTempFile)
+import System.Process
+import qualified Data.Text as Text
+
+main :: IO ()
```

Comment:
Taking the other feedback into account, about only waiting on the port-file, but not also the connection, how significantly does this runner still differ from [`//bazel_tools/client_server/runner`](https://github.com/digital-asset/daml/blob/64abf8aaa9f63386a33182fc6a9e77da60ff0000/bazel_tools/client_server/runner/Main.hs)?
That other runner also supports port 0 and a port-file, it also waits for the creation of that port-file.
One difference that stands out is that the other runner hard-codes the shape of the `--port-file` and `--port` arguments. If that's the only remaining difference, perhaps the other runner could be generalized to make these args configurable instead of adding an additional runner?

Reply:
You're right, they're similar. My goal is to eventually merge all three runners together. However, I wanted to start small and just change a single target.

Reply:
Thanks for clarifying. :+1: Makes sense.